ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -267,17 +335,39 @@ export class AdminWorkspace extends React.Component<{}, State> {
             <PurpleLabel style={styles.narrowWithMargin}>Status</PurpleLabel>
           </FlexRow>
           {resources.clusters.map((cluster, i) =>
-              <FlexRow>
+              <FlexRow key={i}>
                 <div style={styles.narrowWithMargin}>{cluster.clusterName}</div>
                 <div style={styles.narrowWithMargin}>{cluster.googleProject}</div>
                 <div style={styles.narrowWithMargin}>{new Date(cluster.createdDate).toDateString()}</div>
                 <div style={styles.narrowWithMargin}>{new Date(cluster.dateAccessed).toDateString()}</div>
                 <div style={styles.narrowWithMargin}>{cluster.status}</div>
-                <Button>Disable</Button>
+                <Button onClick={() =>
+                  this.setState({clusterToDelete: cluster, confirmDeleteCluster: true})}
+                  disabled={clusterToDelete && clusterToDelete.clusterName === cluster.clusterName}>
```

Comment:
This should also be disabled if the person doesn't have the authority to delete clusters (`SECURITY_ADMIN`).